tests/networkpolicy_test.go

```diff
@@ -176,19 +235,26 @@ func skipNetworkPolicyRunningOnKindInfra() {
 	}
 }
 
-func createVMICirros(virtClient kubecli.KubevirtClient, namespace string, labels map[string]string) *v1.VirtualMachineInstance {
+func createVMICirros(virtClient kubecli.KubevirtClient, namespace string, labels map[string]string, opts ...libvmi.Option) *v1.VirtualMachineInstance {
 	var err error
 	vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
 	vmi.Namespace = namespace
 	vmi.Labels = labels
+
+	// Clean up interfaces since we configure them with `libvmi.Option`
+	vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{}
+
+	for _, opt := range opts {
+		opt(vmi)
+	}
+
 	vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Create(vmi)
 	Expect(err).ToNot(HaveOccurred())
 
 	return vmi
 }
 
 func assertPingSucceed(fromVmi, toVmi *v1.VirtualMachineInstance) {
-	ExpectWithOffset(1, toVmi.Status.Interfaces[0].IPs).NotTo(BeEmpty())
```

Comment:
removing this is even out of this PR context, and also i dont think its good to do as we spoke on other threads.
please reconsider

Reply:
If I understand correctly you found and sriov but related to `IPs` but this is for masquerade also if we have a bug and we don't have IPs this is going to be the least of our problems, I removed this to make stuff consistent.

Reply:
We have masquerade test for it, we are checking network policies here.

Reply:
its a simple sanity / assert, the tests should be self contained in this aspect imo.

Reply:
its a simple sanity / assert, the tests should be self contained in this aspect imo.
no one would remember or should remember what test do which sanity, and its not right imo that one test would do them all / most, and the rest might check air, races can happen and each test has a bit different flow.

Reply:
For masquerade binding this is test here https://github.com/kubevirt/kubevirt/blob/master/tests/vmi_networking_test.go#L721 we cannot duplicate all the assert at all the tests in all the checks, the tests should check only whatever if testing it.

I understand that if we have a problem and there is no IPs we will have a false possitive if we think that this is important let's just go back and put some assert at BeforeEach.

Reply:
i dont see it as duplication, i see it as a simple preliminary check. 

if you check in BeforeEach, and the ips go away until you ping, you wont find it

so i would either not touch it in this PR, or sample it near the ping and iterate + validate the sampled, 
this way it wont be false positive, and the only situation it wont detect is if the ips disappear after the sampling.

its also a good practice to check directly before use, as checking for nil pointer just before use and not on all the call stack (in case a function pass to other function and so on)

Reply:
I am going to preapre again the functions that check them and return the VMIs this is the most we can do we are not going to gain anyghing moving them to ping if IPs appear or disappear we have a flaky and we have to investigate.

Reply:
thought you meant BeforeEach, (this is what you wrote)
i prefer it on BeforeEach if possible ?